### PR TITLE
[ADD] mrp_sale_info

### PR DESCRIPTION
**Pull request by JavierIniesta**
# 
# MRP Sale Info

This module extends the functionality of _sale_mrp_.
- Add fields to Manufacturing Orders and Work Orders:
  - Sale order
  - Customer
  - Commitment Date
# Usage

New fields are in Tree views and in Extra information from form views.

### CODE REVIEW

**Comment by elicoidal** [comment]
LGTM. This module already works as is in v9 AFAIK

File: mrp_sale_info/README.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+* Add fields to Manufacturing Orders and Work Orders:
+    * Sale order
+    * Customer
+    * Commitment Date
+
+Usage
+=====
+
+New fields are in Tree views and in Extra information from form views.
```
Comment (antespi) [inline mrp_sale_info/README.rst]: s/from/tab at/

File: mrp_sale_info/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "version": "8.0.1.0.0",
+    "category": "Manufacturing",
+    "website": "http://www.antiun.com",
+    "author": "Antiun Ingeniería S.L., "
+              "Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    "depends": [
+        "base",
```
Comment (antespi) [inline mrp_sale_info/__openerp__.py]: base addon is not need if depends list is not empty.

**Comment by JavierIniesta** [comment]
Changes done. Thanks!

**Comment by JMODaERP** [comment]
I tested :+1:

**Comment by anajuaristi** [comment]
Hi, 
Sad to see that this module has been developed by "OdooMRP team", this is @pedrobaeza and avanzosc inside the OdooMRP project, and we are not mentioned as authors of it, not even the project.

@rafaelbn I wish it has been an error and you solve it asap,
Thank you:
Ana

**Comment by rafaelbn** [comment]
@JavierIniesta please add author as @anajuaristi say please!. Thanks!

**Comment by rafaelbn** [comment]
After fixing authors, the module is ready to merge :+1:

**Comment by JavierIniesta** [comment]
Authors and contributors sections fixed.

File: mrp_sale_info/models/mrp_production.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+                              compute='_compute_sale_id')
+    partner_id = fields.Many2one(related='sale_id.partner_id',
+                                 string='Customer', store=True)
+    commitment_date = fields.Datetime(related='sale_id.commitment_date',
+                                      string='Commitment Date', store=True)
+
+    @api.one
+    @api.depends('sale_name')
+    def _compute_sale_id(self):
+        sale = self.env['sale.order'].search([('name', '=', self.sale_name)])
```
Comment (oihane) [inline mrp_sale_info/models/mrp_production.py]: There is a direct link between `mrp.production` and `sale.order` that does not depend on the name of the `sale.order`, check this out: https://github.com/odoomrp/odoomrp-wip/blob/8.0/mrp_production_sale_info/models/mrp_production.py#L17

File: mrp_sale_info/models/mrp_production.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    sale_id = fields.Many2one('sale.order', string='Sale order',
+                              readonly=True, store=True,
+                              compute='_compute_sale_id')
+    partner_id = fields.Many2one(related='sale_id.partner_id',
+                                 string='Customer', store=True)
+    commitment_date = fields.Datetime(related='sale_id.commitment_date',
+                                      string='Commitment Date', store=True)
+
+    @api.one
```
Comment (oihane) [inline mrp_sale_info/models/mrp_production.py]: Just to make it easier portable to next version where @api.one is deprecated  `_compute` fields must be declared as @api.multi, and then you have to do a for inside.

**Comment by oihane** [comment]
@JavierIniesta I think that with transifex, there is not needed to add *.pot file, @pedrobaeza am I right?

File: mrp_sale_info/views/mrp_production.xml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+<data>
+
+    <record id="mrp_production_form_view" model="ir.ui.view">
+        <field name="name">MRP Production Form with Sale Order</field>
+        <field name="model">mrp.production</field>
+        <field name="inherit_id" ref="mrp.mrp_production_form_view"/>
```
Comment (oihane) [inline mrp_sale_info/views/mrp_production.xml]:
``` xml
<field name="inherit_id" 
       ref="sale_mrp.mrp_production_form_view_inherit_sale_mrp" />
```

Comment (JavierIniesta) [inline mrp_sale_info/views/mrp_production.xml]: I prefer inherit from the base view.

**Comment by JavierIniesta** [comment]
@oihane  Changes done. Thanks!

File: mrp_sale_info/views/mrp_production.xml (hunk 5 above)
Comment (oihane) [inline mrp_sale_info/views/mrp_production.xml]: My comment was because runbot was failing, as sale_name is not in that view.

**Comment by oihane** [comment]
Travis fails:

```
mrp_sale_info/models/mrp_production.py:5:1: F401 'api' imported but unused
```

File: mrp_sale_info/views/mrp_production.xml (hunk 5 above)
Comment (JavierIniesta) [inline mrp_sale_info/views/mrp_production.xml]: You're right!

**Comment by JavierIniesta** [comment]
Runbot was killed.

**Comment by oihane** [comment]
Runbot is finally green, there is another PR with the module first developed in odoomrp https://github.com/OCA/manufacture/pull/42

File: mrp_sale_info/models/mrp_production.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+from openerp import models, fields
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    sale_id = fields.Many2one(
+        'sale.order', string='Sale order', readonly=True, store=True,
+        related='move_prod_id.procurement_id.sale_line_id.order_id')
+    partner_id = fields.Many2one(related='sale_id.partner_id',
```
Comment (pedrobaeza) [inline mrp_sale_info/models/mrp_production.py:14]: I remember a problem with this transitivity (a related field from another related field). @oihane, do you remember? @JavierIniesta, have you tested this in a real environment?

Comment (oihane) [inline mrp_sale_info/models/mrp_production.py:14]: Sorry, I do not remember, our module is used as information only, those fields are not functionally used for anything else.

Comment (pedrobaeza) [inline mrp_sale_info/models/mrp_production.py:14]: OK, let's merge and see if there are problems

**Pull request merged**